feat(ToolbarMenu): sticky formatting toolbar position - I314

shrey-sachdeva2000 opened the pull request:
Signed-off-by: Shrey Sachdeva <shreysachdeva.2000@gmail.com>

# Issue #314 
Make `ToolbarMenu` sticky

### Changes
- `ToolbarMenu` is now sticky and attaches to the top of the editor on scroll

### Flags
- N/A

irmerk commented:
## As is in `cicero-ui`

![Apr-13-2020 09-14-48](https://user-images.githubusercontent.com/36460856/79123080-4c188400-7d67-11ea-9084-1300bbb2ad6c.gif)

## Adjustments

![Apr-13-2020 09-20-50 NEW](https://user-images.githubusercontent.com/36460856/79123409-1922c000-7d68-11ea-8d9e-04e34e1f6ac9.gif)

Adjustments are to `FormattingToolbar`: `margin: 0 -20px`, `padding: 15px 35px`

irmerk commented:
@Michael-Grover notice the toolbar doesn't cover a clause template in the first gif, and does in the second.

irmerk commented:
This could potentially also use a (bottom) border.

shrey-sachdeva2000 commented:
![ToolbarMenu](https://user-images.githubusercontent.com/47667325/79125399-beb34f00-7dbb-11ea-9637-c6f65cb7ef25.PNG)
@irmerk `margin: 0 -20px` pushes the `ToolbarMenu` slightly out of the `Editor`. Should I change it anyway or adjust the `margin` just so it fits inside?

irmerk commented:
Hmm unsure, we want this to look correct in _both_ `markdown-editor` and `cicero-ui`. This could just be an issue in the `markdown-editor`'s demo?

Michael-Grover commented:
This looks good in the markdown editor but I can't tell whether it would look good in cicero UI

DianaLease commented:
I think the margin issue is more an issue of width. The width of the toolbar should depend on the width of the editor (which may differ depending on where the editor is used). I think the toolbar should always take up the full width of the editor.

shrey-sachdeva2000 commented:
Please review @irmerk @DianaLease @Michael-Grover. I can fix the tests as well if this is okay.

DianaLease commented on examples/src/index.js:
```diff
@@ -131,7 +131,7 @@ function Demo() {
           </Grid.Column>
 
-          <Grid.Column>
+          <Grid.Column className="ap-richtext-editor">
```
this class name is a little confusing since we also have `ap-rich-text-editor` which is only different by a hyphen in the src code. Can you make this one `demo-rich-text-editor` instead and then update the `examples/src/index.css` file as well?

shrey-sachdeva2000 replied:
Sure! Should I just change that with the test update commit?

irmerk commented:
Yeah I think fix  the snapshot tests and we can get a micro-publish from Travis and test this in `cicero-ui` before merging.

shrey-sachdeva2000 commented:
> Yeah I think fix the snapshot tests and we can get a micro-publish from Travis and test this in `cicero-ui` before merging.

Hi @irmerk! Updated the snapshot and squashed the commits. Can you please test this in `cicero-ui`?

irmerk commented:
@shrey-sachdeva2000 I was mistaken, this needs to be a branch for Travis to do a micropublish. Can you open a new PR and point it to a new branch instead of `accordproject:master`? We'll merge it in, Travis will publish a micro npm package, then we can test better in `cicero-ui` and adjust the branch and then once that's good, open a NEW PR that merges that branch into `master`

shrey-sachdeva2000 commented:
Okay sure @irmerk. Can you please create that new branch and I'll point the PR against that. Or maybe we could just point it to the `development` branch.

irmerk commented:
Yes sorry about that, made `shrey-sticky`

irmerk commented:
For reviewers, use: `@accordproject/markdown-editor@0.9.18-20200417202334` in `cicero-ui`

irmerk commented:
I think we need `src/RichTextEditor/FormattingToolbar/index.js` to make `<ToolbarMenu>` have a `className` so that `cicero-ui` can alter it to ensure it overlaps the clauses.

shrey-sachdeva2000 commented:
Yeah that makes sense! Just updated it.

irmerk commented:
> Yeah that makes sense! Just updated it. Do I make a new PR to `shrey-sticky` branch?

Yes please do!

irmerk commented:
For reviewers, use: `@accordproject/markdown-editor@0.9.18-20200420131916 ` in `cicero-ui`

irmerk commented:
@shrey-sachdeva2000 it looks like that works. Maybe change the `className` to `ap-rich-text-editor-toolbar`?

@DianaLease can you test this in the `cicero-ui` demo? Customize the toolbar to make it the right width and margins.

DianaLease commented:
> For reviewers, use: `@accordproject/markdown-editor@0.9.18-20200420131916 ` in `cicero-ui`

For some reason, when I use this in cicero-ui, the 20px of padding on the element with the`ap-rich-text-editor` class doesn't appear in the UI, which causes an issue. Once I manually add this padding using inspector, it looks fine.

shrey-sachdeva2000 commented:
Sure, ready to be merged. Thanks! @irmerk
